autoscheduler/frontend/src/components/LandingPage/SelectTerm/SelectTerm.tsx

```diff
@@ -23,19 +29,38 @@ const getTermsJson = ((): () => Promise<any> => {
   };
 })();
 
-const SelectTerm: React.FC = () => {
+const SelectTerm: React.FC<SelectTermProps> = ({ navBar = false }) => {
+  const dispatch = useDispatch();
   // anchorEl tells the popover menu where to center itself. Null means the menu is hidden
   const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
   const [options, setOptions] = React.useState<string[]>([]);
   const [termMap, setTermMap] = React.useState<Map<string, string>>(new Map());
   const open = Boolean(anchorEl);
 
+  // Used for retrieving the user-friendly term phrase given the term code (e.g. "202031")
+  const [inverseTermMap, setInverseTermMap] = React.useState<Map<string, string>>(new Map());
+  const globalTerm = useSelector<RootState, string>((state) => state.termData.term);
+
+  let styles = defaultStyles;
+
+  if (navBar) {
+    styles = navBarStyles;
+  }
+
   // Fetch all terms to use as ListItem options
   function getTerms(): void {
     getTermsJson().then(
       (res) => {
-        const termsMap = new Map<string, string>(Object.entries(res));
+        const termsMap: Map<string, string> = new Map(Object.entries(res));
+        // Inverse the terms map so that it's in the format of [termNumber: termSentence],
+        // Ex: terms map is like ["Fall 2020 - College Station": 202031],
+        // and inverseTermsMap is [202031, "Fall 2020 - College Station"]
+        const inverseTermsMap: Map<string, string> = new Map(
+          //  a[1] (ex. "202031") is given as an "unknown", so we need to cast it to a string
+          Object.entries(res).map((a) => [String(a[1]), a[0]]),
```

Comment:
```suggestion
          Array.from(termsMap.entries(), ([name, code]) => [code, name]),
```
A little more verbose, but it's more clear what this is going for